torchgeo/transforms/transforms.py

```diff
@@ -87,8 +204,25 @@ def apply_transform(
             padding=flags['padding'],
         )
 
-        if flags['keepdim']:
-            out = rearrange(out, 'b t c h w -> (b t) c h w')
+        # Handle temporal data from VideoSequential
+        # If we extracted multiple patches and batch size suggests temporal data (even number),
+        # rearrange to group patches by spatial location rather than temporal sequence
+        # Only apply this fix when keepdim=True (for flattening compatibility)
+        if (
+            len(out.shape) == 5
+            and out.shape[1] > 1
+            and input.shape[0] % 2 == 0
```

Comment:
I don't think this check here is a good way to test for temporal data. It then fails on masks since they have shape of 1.